File: check_braces.js
const fs = require('fs');

const content = fs.readFileSync(process.argv[2], 'utf8');
let stack = [];
const lines = content.split('\n');

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    for (let j = 0; j < line.length; j++) {
        const char = line[j];
        if (char === '{') {
            stack.push({ char, line: i + 1, col: j + 1 });
        } else if (char === '}') {
            if (stack.length === 0) {
                console.log(`Unexpected } at ${i + 1}:${j + 1}`);
            } else {
                stack.pop();
            }
        }
    }
}

if (stack.length > 0) {
    console.log(`Unclosed braces: ${stack.length}`);
    stack.slice(-10).forEach(s => console.log(`  { at ${s.line}:${s.col}`));
} else {
    console.log("All braces matched!");
}

File: check_braces.py
import sys

def check_braces(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    stack = []
    lines = content.split('\n')
    for line_num, line in enumerate(lines, 1):
        for char_num, char in enumerate(line, 1):
            if char == '{':
                stack.append(('{', line_num, char_num))
            elif char == '}':
                if not stack:
                    print(f"Unexpected '}}' at {line_num}:{char_num}")
                else:
                    stack.pop()
            # Note: This is simplified and doesn't handle strings/comments, 
            # but usually suffices to find large structural issues.
    
    if stack:
        print(f"Unclosed braces: {len(stack)}")
        for b, l, c in stack[-5:]:
            print(f"  '{b}' at {l}:{c}")

if __name__ == "__main__":
    check_braces(sys.argv[1])
